rover_video/src/video_recording.cpp

```diff
@@ -0,0 +1,257 @@
+    _filename(filename_in),
+    _videoFolderPath(videoFolderPath_in),
+    rLogger(logger)
+{
+}
+
+/**
+ * @brief Construct a new Recording:: Recording object using move
+ * @brief Used when emplacing temporary objects into a hashmap
+ *
+ * @param other Recording object
+ */
+Recording::Recording(Recording&& other) noexcept:
+    _RequestShutdown(std::move(other._RequestShutdown)),
+    _recordingThread(std::move(other._recordingThread)),
+    _camURL(std::move(other._camURL)),
+    _pipeline(std::move(other._pipeline)),
+    _filename(std::move(other._filename)),
+    _videoFolderPath(std::move(other._videoFolderPath)),
+    _files(std::move(other._files)),
+    _recordingNumber(other._recordingNumber),
+    _startTime(other._startTime),
+    _frame_width(other._frame_width),
+    _frame_height(other._frame_height),
+    _fps(other._fps),
+    rLogger(std::move(other.rLogger)),
+    _cap(std::move(other._cap)),
+    _video_writer(std::move(other._video_writer)),
+    _appender(std::move(other._appender)),
+    _frame(std::move(other._frame))
+{
+    _stopRecording.store(other._stopRecording.load());  // cannot move atomic
+}
+
+/**
+ * @brief Move operator for the recording class
+ *
+ * @param other Recording object
+ * @return Recording&
+ */
+Recording& Recording::operator=(Recording&& other) noexcept
+{  // move operator just to be safe
+    if (this != &other)
+    {  // Prevent self-assignment
+
+        // Move resources
+        _RequestShutdown = std::move(other._RequestShutdown);
+        _recordingThread = std::move(other._recordingThread);
+        _stopRecording.store(other._stopRecording.load(std::memory_order_acquire), std::memory_order_release);
+
+        _camURL = std::move(other._camURL);
+        _pipeline = std::move(other._pipeline);
+        _filename = std::move(other._filename);
+        _videoFolderPath = std::move(other._videoFolderPath);
+        _files = std::move(other._files);
+        _recordingNumber = other._recordingNumber;
+        _startTime = other._startTime;
+        _frame_width = other._frame_width;
+        _frame_height = other._frame_height;
+        _fps = other._fps;
+        rLogger = std::move(other.rLogger);
+
+        _appender = std::move(other._appender);
+        _cap = std::move(other._cap);  // Move cv ressources
+        _video_writer = std::move(other._video_writer);
+        _frame = std::move(other._frame);
+    }
+    return *this;
+}
+
+/**
+ * @brief Destroy the Recording:: Recording object \n
+ * @brief Handle the release of CV objects
+ *
+ */
+Recording::~Recording(void)
+{
+    if (_cap.isOpened())  // avoid unnecessary logging when creating temporary objects
+    {
+        _stopRecording.store(true);
+
+        if (_recordingThread.joinable())
+        {
+            _recordingThread.join();
+        }
+        // Release resources
+        _cap.release();
+        _video_writer.release();
+        _appender.release();
+
+        RCLCPP_INFO(*rLogger, "Recording stopped.");
```

Comment:
If using a ptr for the logger, you need to make sure it's valid, for example:

```suggestion
        if (rLogger)
        {
            RCLCPP_INFO(*rLogger, "Recording stopped.");
        }
```

Using a reference would make sense in the case, or construct a new one if none is provided